modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -327,39 +384,241 @@ public function insertOrderToDb($id_order, $address, $value, $bits) {
+            $key,
+            OPENSSL_RAW_DATA,
+            $iv
+        );
+    }
+
+    /**
+     * Encrypts a string using the application secret. This returns a hex representation of the binary cipher text
+     */
+    public function encrypt(string $input): string
+    {
+		$encryption_algorithm = 'AES-256-CBC';
+		$hashing_algorith = 'sha256';
+    	$secret = $this->getCallbackSecret();
+        $key = hash($hashing_algorith, $secret, true);
+        $iv = random_bytes(16);
+
+        $cipherText = openssl_encrypt(
+            $input,
+            $encryption_algorithm,
+            $key,
+            OPENSSL_RAW_DATA,
+            $iv
+        );
+        $hash = hash_hmac($hashing_algorith, $cipherText, $key, true);
+
+        return bin2hex($iv . $hash . $cipherText);
+    }
+
 	/*
-	 * Try to get order row from db by address
+	 * Add a new order to the db
 	 */
-	public function getOrderByAddress($bitcoinAddress) {
+	public function newOrder($amount, $id_order) {
+
 		try {
 			$existing_order = Capsule::table('blockonomics_bitcoin_orders')
-				->where('addr', $bitcoinAddress)
+				->where('id_order', $id_order)
+				->where('addr', '')
+				->where('value', $amount)
+				->value('id_order');
+		} catch (\Exception $e) {
+				echo "Unable to select order from blockonomics_bitcoin_orders: {$e->getMessage()}";
+		}
+
+		if(!$existing_order) {
+			try {
+				Capsule::table('blockonomics_bitcoin_orders')->insert(
+					[
+						'id_order' => $id_order,
+						'timestamp' => time(),
+						'status' => -1,
+						'value' => $amount
+					]
+				);
+			} catch (\Exception $e) {
+					echo "Unable to insert new order into blockonomics_bitcoin_orders: {$e->getMessage()}";
+			}
+		}		
+
+		$crypted_id = $this->encrypt($id_order);
+
+		return $crypted_id;
+	}
+
+	/*
+	 * Get the order id using the order uuid
+	 */	
+	public function getOrderIdByHash($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		try {
+			$first_order = Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
 				->orderBy('timestamp', 'desc')
 				->first();
 		} catch (\Exception $e) {
-				echo "Unable to select order from blockonomics_bitcoin_orders: {$e->getMessage()}";
+				echo "Unable to get order id from blockonomics_bitcoin_orders: {$e->getMessage()}";
 		}
 
-		$row_in_array = array(
-			"id" => $existing_order->id,
-			"order_id" => $existing_order->id_order,
-			"timestamp"=> $existing_order->timestamp,
-			"status" => $existing_order->status,
-			"value" => $existing_order->value,
-			"bits" => $existing_order->bits,
-			"bits_payed" => $existing_order->bits_payed
-		);
+		return $first_order->id_order;
+	}
 
-		return $row_in_array;
+	/*
+	 * Get all orders linked to uuid
+	 */	
+	public function getAllOrdersByHash($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		try {
+			return Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
```

Comment:
Order by timestamp is unnecessary as it creates additional overhead on query

Reply:
Removed in 7e4b565